pkg/testcase/secretsencrypt.go

```diff
@@ -26,27 +24,59 @@ func TestSecretsEncryption(cluster *shared.Cluster) {
 
 	index := len(nodes) - 1
 	cpIp := nodes[index].ExternalIP
-	shared.LogLevel("info", "TEST: Old Method of Secrets-Encryption")
-	secretsEncryptOps("prepare", product, cluster.ServerIPs[0], cpIp, nodes)
-	secretsEncryptOps("rotate", product, cluster.ServerIPs[0], cpIp, nodes)
-	secretsEncryptOps("reencrypt", product, cluster.ServerIPs[0], cpIp, nodes)
-	shared.LogLevel("info", "TEST: New Method of Secrets-Encryption")
-	secretsEncryptOps("rotate-keys", product, cluster.ServerIPs[0], cpIp, nodes)
+	seErr := performSecretsEncryption(flags.SecretsEncrypt.Method, product,
+		cluster.Config.ServerFlags, cluster.ServerIPs[0], cpIp, nodes)
+	Expect(seErr).NotTo(HaveOccurred(), "error performing secrets-encryption")
+}
+
+func performSecretsEncryption(method, product, serverFlags, primaryNodeIp, cpIP string, nodes []shared.Node) (err error) {
+	actions := []string{"prepare", "rotate", "reencrypt", "rotate-keys"}
+	switch method {
+	case "reencrypt":
+		shared.LogLevel("info", "Performing secrets-encryption using prepare, rotate and reencrypt...")
+		for _, action := range actions[:len(actions)-1] {
+			err = secretsEncryptOps(action, product, serverFlags, primaryNodeIp, cpIP, nodes)
+			if err != nil {
+				return shared.ReturnLogError("error performing secrets-encrypt %s operation on node: %s!", action, cpIP)
+			}
+		}
+	case "rotate-keys":
+		shared.LogLevel("info", "Performing secrets-encryption using rotate-keys...")
+		action := actions[len(actions)-1]
+		err = secretsEncryptOps(action, product, serverFlags, primaryNodeIp, cpIP, nodes)
+		if err != nil {
+			return shared.ReturnLogError("error performing secrets-encrypt %s operation on node: %s!", action, cpIP)
```

Comment:
Can we instead return "% secret-encrypt %s on node: %s failed", product, action, cpIP?  Provides clarity on what failed instead of notes format.  Can we log the actual err?